.github/workflows/pkgci_regression_test.yml

```diff
@@ -95,6 +95,13 @@ jobs:
         if: "!cancelled()"
         run: |
           source ${VENV_DIR}/bin/activate
+          case "${{ matrix.name }}" in
+            "amdgpu_rocm_mi300_gfx942") IREE_TEST_FILES="/shark-cache/data/iree-regression-cache" ;;
+            *) echo "No cache directory assigned for ${{ matrix.name }}" ;;
+          esac
+          if [[ -n "$IREE_TEST_FILES" ]]; then
+            export IREE_TEST_FILES
+          fi
```

Comment:
Rather than copy/paste this into three steps, add a step that writes to GITHUB_ENV

* https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#passing-values-between-steps-and-jobs-in-a-workflow
* https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable

I saw some commits on your other PR trying variations on that. In general, prefer to debug on smaller workflows (possibly on a fork) instead of triggering the full production CI. Each run has material costs associated with it.